crates/storage/store.rs

```diff
@@ -42,7 +42,28 @@ pub enum EngineType {
     RedB,
 }
 
+pub struct UpdateBatch {
+    /// Nodes to be added to the state trie
+    pub account_updates: Vec<TrieNode>,
+    /// Storage tries updated and their new nodes
+    pub storage_updates: Vec<(H256, Vec<TrieNode>)>,
+    /// Blocks to be added
+    pub blocks: Vec<Block>,
+    /// Receipts added per block
+    pub receipts: Vec<(H256, Vec<Receipt>)>,
+}
+
+impl UpdateBatch {
+    pub async fn apply_to_store(self, store: Store) -> Result<(), StoreError> {
```

Comment:
what's the purpose of this method? can't we just call the store method directly?

Reply:
ok, I've removed the method